### ISSUE TITLE
Response should still be returned for integration testing extensions

### ISSUE THREAD

**Issue by austins** [FastEndpoints/FastEndpoints#425]
The HttpClient extensions for integration testing throws a InvalidOperationException if the response cannot be deserialized, and the response isn't returned.

This is thrown if the endpoint doesn't have AllowAnonymous. For testing, it should return the response so the status code can be asserted as the actual app does return a response.

One workaround to assert the status code is checking if the exception message contains `[Unauthorized]` in this case. Example:
```
        // Act
        var act = async () =>
            await HttpClient.GETAsync<GetArticleByIdEndpoint, GetArticleByIdRequest, ArticleDto>(request);

        // Assert
        await act.Should().ThrowAsync<InvalidOperationException>().WithMessage($"*[{HttpStatusCode.Unauthorized}]*");
```

Another workaround is to set the response generic param to `EmptyResponse` on the extension method. It doesn't seem right as the endpoint when the app is running is defined with the response in the generic there.

I presume this issue would also occur if the endpoint handler calls `await SendUnauthorizedAsync(); return;` or `SendNotFoundAsync()` etc. before returning a successful response.

```
        try
        {
            res = await rsp.Content.ReadFromJsonAsync<TResponse>(SerOpts.Options);
        }
        catch (JsonException)
        {
            var reason = $"[{rsp.StatusCode}] {await rsp.Content.ReadAsStringAsync()}";
            throw new InvalidOperationException(
                $"Unable to deserialize the response body as [{typeof(TResponse).FullName}]. Reason: {reason}");
        }

        return new(rsp, res);
```

**Comment by austins** [comment]
Never mind, I see this was fixed just yesterday with #422. 👍 
https://github.com/FastEndpoints/FastEndpoints/blob/cea5b3526c59dd8836e7b462e1cc025116d71f44/Src/Library/Testing/HttpClientExtensions.cs#L226-L242

**Comment by dj-nitehawk** [comment]
@austins 

decided to do a breaking change and finally sort this out:
https://github.com/FastEndpoints/FastEndpoints/blob/main/Src/Library/changelog.md#breaking-changes

available in `v5.8.1.10-beta`

**Comment by austins** [comment]
@dj-nitehawk Just tested out the new change. It works beautifully. Thank you.
